examples/auth-hook.py

```diff
@@ -0,0 +1,33 @@
+# Example how to roll your own Authorization in the dredd hooks
+import json
+import dredd_hooks as hooks
+import requests
+
+stash = {}
+api_username = "XXX"
+api_password = "XXX"
+api_uri = "XXX"
+
+
+@hooks.before_each
+def add_token(transaction):
+    if 'token' in stash:
+        print('adding a token')
+    else:
+        stash['token'] = obtain_token()
+
+    transaction['request']['headers']['Authorization'] = "Bearer " + stash['token']
+
+
+def obtain_token():
+    data = {
+               "api_login_form": {
+                   "username": api_username,
+                   "password": api_password
+               }
+           }
+    url = api_uri
+    response = requests.post(url, json=data)
+    parsed_body = json.loads(response.text)
```

Comment:
there is `response.json()` that will return the parsed body. You won't need to manually parse the JSON body.